resource/systemd/unit/unit_test.go

```diff
@@ -0,0 +1,273 @@
+	_, err = linked.Apply()
+	assert.NoError(t, err)
+	status, err := linked.Check(&fr)
+	assert.NoError(t, err)
+	assert.Equal(t, resource.StatusNoChange, status.StatusCode())
+	assert.Contains(t, status.Messages(), fmt.Sprintf("property \"UnitFileState\" of unit %q is \"linked\", expected one of [\"linked\"]", svc.Name))
+	assert.False(t, status.HasChanges())
+
+	// Testing disabling
+	disabled := unit.Unit{Name: svc.Name, Active: false, UnitFileState: "disabled", StartMode: "replace"}
+	_, err = disabled.Apply()
+	assert.NoError(t, err)
+	status, err = disabled.Check(&fr)
+	assert.NoError(t, err)
+	assert.Equal(t, resource.StatusWontChange, status.StatusCode())
+	assert.Contains(t, status.Messages(), fmt.Sprintf("unit %q does not exist, considered disabled", svc.Name))
+	assert.False(t, status.HasChanges())
+}
+
+// HelloUnit is a simple service file that is not static
+const HelloUnit = `
+[Unit]
+Description=Foo hello world
+[Service]
+ExecStart=/bin/bash -c "while true; do /bin/echo HELLO WORLD; sleep 5; done;"
+
+[Install]
+WantedBy=multi-user.target
+`
+
+// HelloUnitStatic is a simple service file that is  static
+const HelloUnitStatic = `
+[Unit]
+Description=Foo hello world
+[Service]
+ExecStart=/bin/bash -c "while true; do /bin/echo HELLO WORLD; sleep 5; done;"
+`
+
+// TmpService is the information for a temporary service file
+type TmpService struct {
+	Path string
+	Name string
+}
+
+// Remove removes this service file from the system entirely
+func (t *TmpService) Remove() {
+	base := filepath.Base(t.Path)
+
+	disable := "systemctl disable " + base
+	daemonReload := "systemctl daemon-reload"
+	resetFailed := "systemctl reset-failed"
+
+	generator := &shell.CommandGenerator{Interpreter: "/bin/bash"}
+	generator.Run(disable)
+
+	os.Remove(t.Path)
+	locations := []string{"/run/systemd/system/", "/etc/systemd/system/", "/usr/lib/systemd/system/"}
+	for _, l := range locations {
+		os.Remove(filepath.Join(l, base))
+	}
+
+	generator.Run(daemonReload)
+	generator.Run(resetFailed)
+}
+
+var count uint32
+
+// NewTmpService creates a service file on the system either in
+// "/tmp", "/etc/systemd/system", or "/run/systemd/system".
+func NewTmpService(prefix string, static bool) (svc *TmpService, err error) {
+	atomic.AddUint32(&count, 1)
+	name := fmt.Sprintf("foo%d.service", count)
+	var path string
+	if prefix == "/tmp" {
+		path = filepath.Join(prefix, name)
+	} else {
+		path = filepath.Join(prefix, "systemd/system", name)
+	}
+
+	if static {
+		err = ioutil.WriteFile(path, []byte(HelloUnitStatic), 0777)
+	} else {
+		err = ioutil.WriteFile(path, []byte(HelloUnit), 0777)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return &TmpService{Path: path, Name: name}, systemd.ApplyDaemonReload()
+}
+
+// IsRoot checks if the current user is Root
+func IsRoot() bool {
+	currentUser, _ := user.Current()
+	return currentUser.Uid == "0"
+}
+
+// HasSystemd checks if dbus is available
+func HasSystemd() bool {
```

Comment:
It's a bit concerning that none of the unit tests can run on a system without systemd available.  Have you considered possibly seeing if there are any tests that you could run without having systemd available.  Usually unit tests should be able to run on every architecture that we plan on supporting, and integration tests should run on the relevant platforms.  This way if someone changes a higher level API that might break the module they'll be able to tell even if they're developing on a system without systemd.

It might be worth looking into either mocking the systemd library or breaking apart some of the functionality into smaller units so that you can at least have some test coverage for systems without systemd available.